src/components/multi-factor-authentication/tests/multi-factor-authentication-controller.test.ts

```diff
@@ -0,0 +1,67 @@
+import { expect } from "chai";
```

Comment:
probably worth adding a integration test to verify the validation works correctly and state machine.